pandora-client-web/src/components/settings/securitySettings.tsx

```diff
@@ -0,0 +1,273 @@
+import React, { ReactElement } from 'react';
+import { AssertNever, FormatTimeInterval, PasswordSchema, IDirectoryAccountInfo, IClientDirectoryNormalResult } from 'pandora-common';
+import { useForm } from 'react-hook-form';
+import { useCurrentAccount, useDirectoryConnector } from '../gameContext/directoryConnectorContextProvider';
+import { PrehashPassword } from '../../crypto/helpers';
+import { Form, FormCreateStringValidator, FormField, FormFieldError } from '../common/form/form';
+import { Button } from '../common/button/button';
+import { FieldsetToggle } from '../common/fieldsetToggle/fieldsetToggle';
+import { useAsyncEvent } from '../../common/useEvent';
+import { ModalDialog, useConfirmDialog } from '../dialog/dialog';
+import { useObservable } from '../../observable';
+import type { AuthToken } from '../../networking/directoryConnector';
+import { useCurrentTime } from '../../common/useCurrentTime';
+import { useKeyDownEvent } from '../../common/useKeyDownEvent';
+import { Row } from '../common/container/container';
+
+export function SecuritySettings(): ReactElement | null {
+	const account = useCurrentAccount();
+
+	if (!account)
+		return <>Not logged in</>;
+
+	return (
+		<>
+			<ConnectedClients />
+			<PasswordChange account={ account } />
+		</>
+	);
+}
+
+type AccountConnectedClients = IClientDirectoryNormalResult['queryConnections']['connections'];
+
+function ConnectedClients(): ReactElement {
+	const directoryConnector = useDirectoryConnector();
+	const [connections, setConnections] = React.useState<AccountConnectedClients | null>(null);
+
+	const [load, processing] = useAsyncEvent(
+		async () => await directoryConnector.awaitResponse('queryConnections', {}),
+		(resp) => setConnections(resp.connections),
+	);
+
+	const onChange = React.useCallback((open: boolean) => {
+		if (open && connections === null) {
+			void load();
+		}
+	}, [connections, load]);
+
+	return (
+		<FieldsetToggle legend='Connected Clients' open={ false } onChange={ onChange }>
+			<div>
+				<ConnectedClientsList connections={ connections } />
+				<Button onClick={ load } disabled={ processing }>Refresh</Button>
```

Comment:
The refresh button should be at the top, so user doesn't have to potentially scroll to find it. Also use `Column` instead of simple `div` to get nice spacing between the two elements.